src/ckb-daemon/ckb-daemon.pro

```diff
@@ -14,10 +14,10 @@ macx {
 }
 
 QMAKE_CFLAGS  += -std=gnu11 -Wall -Wextra -fsigned-char
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.9
```

Comment:
You've removed the deployment target everywhere, this will lead to a non-functional package if we decide to cut the release on `master` before cmake substitutes qmake. I mean, if one builds the `.app` on, say, 10.12, then it will only work on 10.12+. This is undesirable. Is it possible to introduce the deployment target back? Unfortunately, I can't test it, however, you can look at the compilation flags, in particular `--mmacosx-version-min=` defines it.

Reply:
The whole point of this change is for QMAKE_MACOSX_DEPLOYMENT_TARGET to work from the command line, etc.   Which it now does, so set it however you want at the root project level.

> this will lead to a non-functional package if we decide to cut the release on master before cmake substitutes qmake. I mean, if one builds the .app on, say, 10.12, then it will only work on 10.12+.

This works, and is correct for qmake, but if you're not sure why don't just pull this branch down, build it, confirm that.

I'm doing this, putting this change together, to be a nice guy, because I need to look into a particular bug in ckb and I can't fix that with the jacked up build process that involves modifying the compiler instead of just correctly configuring the project. Others such as @danybmx, @tatokis have confirmed that this builds properly now; you or your team can confirm likewise.  I don't work for you, and your assumption that I do makes this the end of my unpaid effort.

Tbh, I don't care whether you accept this PR or not, and if not, that's also fine, macOs folks like @danybmx and others who like to work on project where we actually use the tools properly, read the documentation, and correctly configure our projects, are free to come contribute on my fork as they see fit or keep hobbling along here.

Reply:
@sterpe Not sure exactly what went wrong here, or why you are upset.

From what I can see, @light2yellow was in no way insulting you, or being negative towards you.
On the contrary, you are the one with the attitude, making bold claims that we do not read the documentation, and do not "actually use the tools".
It would've been appreciated if instead, you gave us a link to specific parts of the documentation that explain what went wrong, in a broken build system that we inherited from the previous developer.
Documentation or not, having qmake passing a blank `--mmacosx-version-min=` to the compiler is not considered acceptable under any circumstance.

If you do believe that PRs should not be reviewed, or have questions asked about them, then you are most likely in the wrong place, as the point of them is code submission for review and inclusion.
In the end, we will be the ones who have to maintain it if something goes wrong, not you; and in order for that to happen, we need to fully understand it, not contribute more to the already unknown codebase.


>Others such as @danybmx, @tatokis have confirmed that this builds properly now

I made no such claim. I do not use macOS, and neither does @light2yellow. I try to support it when I have free time and only whenever possible.


>you or your team can confirm likewise.

Not sure exactly what you are on about. @light2yellow is part of my/our team, and for a reason.


>I'm doing this, putting this change together, to be a nice guy

We are all here spending our free time on this, to be nice guys, and help those who want their devices to work on their platform of choice despite vendor support; yet people are constantly complaining about it, a lot of times in an inappropriate manner.
Don't really know what you're getting at.

Now, on to the PR itself.
I had a look, and while I can understand the use of variables to populate the plist, which is a welcome change, they are unrelated to getting the project to compile.
I do, however, believe that `QMAKE_MACOSX_DEPLOYMENT_TARGET` should at least be specified in qmake-auto.

Regarding qmake itself: Not sure if you are aware, but sooner or later qmake is going to be replaced with cmake.
There's already a PR open to keep track of it, #281. If you are okay with your fixes being replaced by cmake in the near future, then that's okay with me.
It's up to you what you want to do.

--

@light2yellow, since qmake apparently provides built in variable substitution for the plist, and the ICON variable is defined, there should be no problem.
I can't find any documentation regarding SHORT_VERSION though, so I am not sure if it is predefined. https://doc.qt.io/qt-5.9/qmake-variable-reference.html#qmake-info-plist

Reply:
@sterpe 

I re-read the PR message and did not get from the first glance (and it was a glance, since my time and attention distribution is not prioritized for community-driven, 4-fun projects) the way `QMAKE_MACOSX_DEPLOYMENT_TARGET` is supposed to be used since your changes, nor gave I enough attention to the message itself. So, I can see why there could have been a negative reaction due to my questions being somewhat silly in the context of provided description in the PR message, but your way of communication is unacceptable. Instead of creating a drama out of nowhere you should have [patiently] explained it once again or cite yourself if you've already explained it. If you are not ready to provide sufficient details and/or documentation links for other people to furtherly maintain your "unpaid effort", do not contribute to community-driven projects, because without an explanation in case of misunderstanding and with the attitude a la RTFM your "unpaid efforts" will never be accepted or even looked at.

You are free to talk like that to your Facebook colleagues, but on GitHub, please, try to restrain  yourself when writing a message. That's to address your "I do not work for you". No idea what caused this assumption. Obviously, nobody works for anyone here, this is not a corporation.

Anyway, @tatokis has summarized it well so I'm not going to repeat it. I would only add that you should learn to behave.

Now, on the topic.

* I do believe `VERSION = 0.2.8` shouldn't be there as this is a duplication. As I said earlier, we already use `VERSION` file.
* Since `VERSION = 0.2.8` is not defined, I suppose `@SHORT_VERSION@` won't work. And yes, I still expect you to explain how and why this works. Moreover, I see no reason for it at all. Moreover, I see no reason why this is on a PR called "Fix #232 for Qt 5.9.x", as it has no connection. This PR is treated as a hotfix for something being broken, no new features are expected to be added.
* `ICON` is fine, if it works - okay. But again, why on this PR? How does it "fix" the build?
* `QMAKE_MACOSX_DEPLOYMENT_TARGET` has to be defined somewhere in order to be backwards compatible with the project before this PR. And no, I do not like the idea of doing it through an environment variable.

If you don't mind, I'll pick the relevant changes, merge them and hope for the best [== it compiles on macOS]. If you want something else, I am waiting on this until Europe's Monday.